### PR TITLE
Code Cleanup

### LINKED ISSUE #76: IncludeWhatYouUse
We have a lot of unused and historic includes in the main files.

### PR DESCRIPTION
**Pull request by beetleskin**
Fixes #76 and #75

**Test Plan:**
- find missing violations of http://wiki.ros.org/CppStyleGuide (might be easier with a clang-tidy config we could also use instead of my manual refactoring? maybe as a next step ..)
- confirm no regression in functionality 
- confirm compilation on clean docker (we need a CI .. ;))

### CODE REVIEW

**Comment by schra** [comment]
I'll look at this once we have merged your other PRs :)

**Comment by schra** [comment]
Just as a reminder from #72, could you maybe add this small change in this PR too?

> When clearing the topic input, the error will be "No map received yet" - maybe we should change that to "No NavSatFix message received yet"?

**Comment by beetleskin** [comment]
lets do that after review/concerns fixed - just remind me ;)

**Comment by beetleskin** [comment]
#77 is also part of this PR, I'll rebase after #77 is merged

**Comment by beetleskin** [comment]
@schra are we using a fork of https://github.com/davetcoleman/roscpp_code_format ? what about clang-format and camelCase vs. under_score? Can one set that in the config?

File: src/aerialmap_display.cpp [hunk 1]
```diff
@@ -592,19 +589,23 @@ void AerialMapDisplay::assembleScene()
...
 
   checkRequestErrorRate();
 }
 
 void AerialMapDisplay::transformTileToMapFrame()
 {
-  if (not ref_fix_)
+  if (not ref_fix_ or not center_tile_)
   {
     ROS_FATAL_THROTTLE_NAMED(2, "rviz_satellite", "ref_fix_  not set, can't create transforms");
```
Comment (schra) [inline src/aerialmap_display.cpp:610]: `ref_fix_  not` -> `ref_fix_ not`

File: src/aerialmap_display.h [hunk 2]
```diff
@@ -143,52 +150,45 @@ protected Q_SLOTS:
...
 
+  /// the alpha value of the tile's material
   float alpha_;
+  /// determines which render queue to use
   bool draw_under_;
+  /// the URL of the tile server to use
   std::string tile_url_;
+  /// the zoom to use (Mercator)
   int zoom_;
+  /// the number of adjacent tile blocks
```
Comment (schra) [inline src/aerialmap_display.h:176]: Hm, I think this is too vague. I would say "the number of tiles loaded in each direction around the center tile"

**Comment by schra** [comment]
> @schra are we using a fork of https://github.com/davetcoleman/roscpp_code_format ? what about clang-format and camelCase vs. under_score? Can one set that in the config? Or is that only something clang-tidy can do? Should we add a clang-tidy config? Is there one for ROS? I only know the one from moveit (https://github.com/ros-planning/moveit/blob/master/.clang-tidy).

That's a really good point. I'll add a remark about this in #78 And yeah, I think you can only do that with clang-tidy :)

File: src/aerialmap_display.cpp [hunk 3]
```diff
@@ -329,7 +319,7 @@ void AerialMapDisplay::createTileObjects()
   for (int block = 0; block < (2 * blocks_ + 1) * (2 * blocks_ + 1); ++block)
   {
     // generate an unique name
-    static int count = 0;
+    static size_t count = 0;
```
[the remark this answers is not in the document]
Comment (beetleskin) [reply src/aerialmap_display.cpp:322]: a count is never negative, and `size_t` is `unsigned int`, but better; its encouraged to use for any counter or size: https://en.cppreference.com/w/cpp/types/size_t

File: src/aerialmap_display.cpp [hunk 4]
```diff
@@ -406,16 +407,17 @@ void AerialMapDisplay::requestTileTextures()
...
   {
     setStatus(StatusProperty::Error, "Message", "No NavSatFix received yet");
     return;
   }
 
   try
   {
-    tileCache_.request({ *lastCenterTile_, blocks_ });
-    dirty_ = true;
+    ROS_DEBUG_NAMED("rviz_satellite", "Requesting new tile images from the server");
```
[the remark this answers is not in the document]
Comment (beetleskin) [reply src/aerialmap_display.cpp:418]: how often does it come when you move around? it should only be hit when the center tile is updated, and for that you have to drive quite a bit (or change some params like a maniac)

File: src/aerialmap_display.cpp [hunk 5]
```diff
@@ -577,37 +589,23 @@ void AerialMapDisplay::assembleScene()
...
 
   // since not all tiles were loaded yet, this function has to be called again
-  if (!loadedAllTiles)
+  if (!all_tiles_loaded)
   {
     dirty_ = true;
   }
+  else
+  {
+    ROS_DEBUG_NAMED("rviz_satellite", "Finished assembling all tiles");
```
[the remark this answers is not in the document]
Comment (beetleskin) [reply src/aerialmap_display.cpp:598]: I think the whole algo is not trivial due to its asyc nature. If anything goes wrong, the order counts. And as for the other debug log: this should not hit too often. Lastly, DEBUG logging is disabled by default.

File: src/aerialmap_display.cpp [hunk 6]
```diff
@@ -285,7 +286,7 @@ void AerialMapDisplay::updateTopic()
 void AerialMapDisplay::clearAll()
 {
   ref_fix_ = nullptr;
-  lastCenterTile_ = boost::none;
+  center_tile_ = boost::none;
   destroyTileObjects();
 
   setStatus(StatusProperty::Warn, "Message", "No map received yet");
```
[the remark this answers is not in the document]
Comment (beetleskin) [reply src/aerialmap_display.cpp:292]: done

File: src/mercator.h [hunk 7]
```diff
@@ -43,3 +43,21 @@ inline int zoomToMaxTiles(int zoom)
 {
   return (1 << zoom) - 1;
 }
+
+/**
+ * Calculate the tile width/height in meter
+ */
+inline double getTileWH(double const latitude, int const zoom) {
+  // this constant origins from how the base resolution is calculated;
+  // TODO: actually this not needed and could be removed from both formulas, since they cancel out each other;
```
[the remark this answers is not in the document]
Comment (beetleskin) [reply src/mercator.h:52]: check, although the same URL is present in the callee func

File: src/aerialmap_display.cpp (hunk 1 above)
Comment (beetleskin) [reply src/aerialmap_display.cpp:610]: I'm not getting that :/

File: src/aerialmap_display.h (hunk 2 above)
Comment (beetleskin) [reply src/aerialmap_display.h:176]: ok

**Comment by beetleskin** [comment]
> You missed two variables ;)

inferior humans .. 

I rebased, and fixed concerns - the new commit should be merged into the previous one; but I'd keep the others as they are

File: src/aerialmap_display.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (schra) [reply src/aerialmap_display.cpp:322]: >  its encouraged to use for any counter or size

Well, it doesn't exactly encourage it. It only states that it's typically used for looping.

Here are two reputable resources arguing against size_t (aka an unsigned integer type). One of them actually claims the exact opposite of your statement, that is that unsigneds were a good idea for counts.

* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es106-dont-try-to-avoid-negative-values-by-using-unsigned
* https://google.github.io/styleguide/cppguide.html#Integer_Types

I think that it doesn't really matter in this case too much. So if you *really* want a size_t, I'll accept it anyway.

File: src/aerialmap_display.cpp (hunk 1 above)
Comment (schra) [reply src/aerialmap_display.cpp]: yeah my input was really badly worded. fixed it myself :)

File: src/aerialmap_display.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (beetleskin) [reply src/aerialmap_display.cpp:322]: uff .. as you wish

**Comment by schra** [comment]
LGTM now, thanks again :)

**Pull request merged**